src/semgrep_agent/main.py

```diff
@@ -320,7 +329,7 @@ def protected_main(
         meta.base_commit_ref,
         meta.head_ref,
         semgrep.get_semgrepignore(sapp.scan.ignore_patterns),
-        sapp.is_configured,
+        rewrite_rule_ids and not sapp.is_configured,
```

Comment:
Should we allow this to be set to false if sapp.is_configured is true? i.e. should this be an or?

Reply:
I don't think so - if they are using semgrep cloud and try to rewrite rule ids, semgrepignore will no longer work. Since all their rules will be in one file, there is strictly no benefit to having no-rewrite-rule-ids set

Reply:
Perhaps that means I should change the flag to not let them set --rewrite-rule-ids to true ever, but I didn't like how many double-negatives that put into the logic. It made it hard to reason about.

Reply:
So if we always want rules from semgrep-cloud to rewrite rule_ids shouldn't this be `rewrite_rule_ids or not sapp.is_configured`?

Reply:
We do not always want rules from semgrep-cloud to rewrite rule_ids. We actually _never_ want rules from semgrep-cloud to rewrite rule_ids because rules are downloaded and saved to a flat config file - there is no meaningful path to prefix the rules with. The logic is:

if `sapp.is_configured` is true, then `not sapp.is_configured` is false, so anything anded with `not sapp.is_configured` is also false, so rule ids will never be rewritten ✅

Reply:
Meanwhile, if `sapp.is_configured` is false, then `not sapp.is_configured` is true, so then we follow whatever is specified by the commandline flag ✅

Reply:
Ah makes sense. I got confused with all the nots and rewrite-not-rewrite :D

Reply:
Yeah, it is confusing 🙁 but I couldn't figure out how to write this in a less-confusing way given how semgrep already does it